src/borg/archiver.py

```diff
@@ -1598,7 +1598,9 @@ def do_recreate(self, args, repository, manifest, key, cache):
             if recreater.is_temporary_archive(name):
                 self.print_error('Refusing to work on temporary archive of prior recreate: %s', name)
                 return self.exit_code
-            recreater.recreate(name, args.comment, target)
+            if not recreater.recreate(name, args.comment, target):
+                self.print_error('Nothing to do. Archive was not processed.\n'
```

Comment:
No, if you've specified "--target XYZ" or something like that, then the operation would not complete. Thus, its an error.